src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -1681,7 +1681,7 @@ void Generator::GenerateProvides(const GeneratorOptions& options,
       if (options.import_style == GeneratorOptions::kImportCommonJsStrict) {
         string namespaceObject = *it;
         // Remove "proto." from the namespace object
-        GOOGLE_CHECK(namespaceObject.compare(0, 6, "proto."));
+        GOOGLE_CHECK(!namespaceObject.compare(0, 6, "proto."));
         namespaceObject.erase(0, 6);
```

Comment:
If there is a check for that, do we still need to erase it explicitly?

Reply:
Sorry but what do you mean by "erase it explicitly"? Do you mean change to GOOGLE_CHECK for erasing?